x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/audit/logfile/CapturingLogger.java

```diff
@@ -10,24 +10,30 @@
 import org.apache.logging.log4j.Logger;
 import org.apache.logging.log4j.core.LogEvent;
 import org.apache.logging.log4j.core.LoggerContext;
+import org.apache.logging.log4j.core.StringLayout;
 import org.apache.logging.log4j.core.appender.AbstractAppender;
 import org.apache.logging.log4j.core.config.Configuration;
 import org.apache.logging.log4j.core.config.LoggerConfig;
 import org.apache.logging.log4j.core.filter.RegexFilter;
 import org.elasticsearch.common.logging.ESLoggerFactory;
 import org.elasticsearch.common.logging.Loggers;
 
+import java.nio.charset.StandardCharsets;
 import java.util.ArrayList;
 import java.util.List;
 
 public class CapturingLogger {
 
     public static Logger newCapturingLogger(final Level level) throws IllegalAccessException {
+        return newCapturingLogger(level, null);
+    }
+
+    public static Logger newCapturingLogger(final Level level, StringLayout layout) throws IllegalAccessException {
```

Comment:
`LoggingAuditTrailTests` now assume that the layout for the `CapturingLogger` is JSON (not strictly a requirement, but it's better to check for events in tests as they would be generated outside the test env), yet `CapturingLogger` is used in other places and we don't want to impose another layout everywhere.